wa/workloads/sysbench/__init__.py

```diff
@@ -0,0 +1,215 @@
+                  '''),
+    ]
+
+    def validate(self):
+        if (self.max_requests is None) and (self.max_time is None):
+            self.max_time = 30
+        if self.max_time and (self.max_time + 10) > self.timeout:
+            self.timeout = self.max_time + 10
+        if self.test == 'fileio' and not self.file_test_mode:
+            self.logger.debug('Test is "fileio" and no file_test_mode specified -- using default.')
+            self.file_test_mode = 'seqwr'
+        elif self.test != 'fileio' and self.file_test_mode:
+            raise ConfigError('file_test_mode must not be specified unless test is "fileio"')
+
+    @once
+    def initialize(self, context):
+        exe = Executable(self, self.target.abi, 'sysbench')
+        host_binary = context.resolver.get(exe, strict=False)
+        Sysbench.target_binary = self.target.install(host_binary)
+
+    def setup(self, context):
+        self.host_results_file = None
+        params = dict(test=self.test,
+                      num_threads=self.num_threads)
+        if self.max_requests:
+            params['max_requests'] = self.max_requests
+        if self.max_time:
+            params['max_time'] = self.max_time
+        self.target_results_file = self.target.get_workpath('sysbench_result.txt')
+        self.command = self._build_command(**params)
+
+    def run(self, context):
+        self.target.execute(self.command, timeout=self.timeout)
+
+    def extract_results(self, context):
+        self.host_results_file = os.path.join(context.output_directory, 'sysbench_result.txt')
+        self.target.pull(self.target_results_file, self.host_results_file)
+        context.add_artifact('sysbench_output', self.host_results_file, kind='raw')
+
+    def update_output(self, context):
+        if not self.host_results_file:
+            self.logger.warning('No results file found.')
+            return
+
+        with open(self.host_results_file) as fh:
+            find_line_with('General statistics:', fh)
+            extract_metric('total time', fh.next(), context.output)
+            extract_metric('total number of events', fh.next(), context.output, lower_is_better=False)
+            find_line_with('response time:', fh)
+            extract_metric('min', fh.next(), context.output, 'response time ')
+            extract_metric('avg', fh.next(), context.output, 'response time ')
+            extract_metric('max', fh.next(), context.output, 'response time ')
+            extract_metric('approx.  95 percentile', fh.next(), context.output)
+            find_line_with('Threads fairness:', fh)
+            extract_threads_fairness_metric('events', fh.next(), context.output)
+            extract_threads_fairness_metric('execution time', fh.next(), context.output)
+
+    def teardown(self, context):
+        self.target.remove(self.target_results_file)
+
+    @once
+    def finalize(self, context):
+        self.target.uninstall(Sysbench.target_binary)
```

Comment:
Should this be `self.target.uninstall('sysbench')` instead of providing the full path?

Reply:
Updated.